File: app/src/main/java/org/gears3d/gears3d/Gears3dRenderer.java
package org.gears3d.gears3d;

import android.opengl.GLES20;
import android.opengl.GLSurfaceView;

import javax.microedition.khronos.egl.EGLConfig;
import javax.microedition.khronos.opengles.GL10;

public class Gears3dRenderer implements GLSurfaceView.Renderer {
    Gears3d mGears3d;

    public Gears3dRenderer() {
        mGears3d = new Gears3d();
    }

    @Override
    public void onSurfaceCreated(GL10 gl10, EGLConfig eglConfig) {
        mGears3d.set_global_state();
    }

    @Override
    public void onSurfaceChanged(GL10 gl10, int width, int height) {
        mGears3d.win_resize(width, height);
    }

    @Override
    public void onDrawFrame(GL10 gl10) {
        mGears3d.draw();
    }
}

File: app/src/main/kotlin/org/gears3d/gears3d/Gears3dRenderer.kt
package org.gears3d.gears3d

import android.opengl.GLSurfaceView
import javax.microedition.khronos.egl.EGLConfig
import javax.microedition.khronos.opengles.GL10

class Gears3dRenderer : GLSurfaceView.Renderer {
    var mGears3d: Gears3d

    init {
        mGears3d = Gears3d()
    }

    override fun onSurfaceCreated(gl10: GL10, eglConfig: EGLConfig) {
        mGears3d.set_global_state()
    }

    override fun onSurfaceChanged(gl10: GL10, width: Int, height: Int) {
        mGears3d.win_resize(width, height)
    }

    override fun onDrawFrame(gl10: GL10) {
        mGears3d.draw()
    }
}
